src/toil/cwl/cwltoil.py

```diff
@@ -1805,6 +1806,77 @@ def determine_load_listing(tool: ToilCommandLineTool):
         )
     return load_listing
 
+class NoAvailableJobStoreException(Exception):
+    """Indicates that no job store name is available."""
+    pass
+
+def generate_default_job_store(batch_system_name: Optional[str], provisioner_name: Optional[str],
+                               local_directory: str) -> str:
```

Comment:
Does a `local` directory input make sense when asking for a jobstore for `aws` or `gcp`?

Reply:
I need the local directory input because the logic to determine if we're asking for a job store for AWS or GCP, or whether it would be appropriate to actually use the local directory, lives inside the function; I wouldn't know whether to pass it along or not without duplicating that logic.

All the logic to determine what `workdir` should be is in `main`, and it's complicated and hard to move safely. I think that, when we're inventing a local job store, we have to have it equal to `workdir`. I don't really want to make the function `generate_default_job_store_and_determine_workdir()`.